packages/jsapi-components/src/useSelectDistinctTable.ts

```diff
@@ -0,0 +1,34 @@
+import { useCallback, useEffect } from 'react';
+import { Table, TreeTable } from '@deephaven/jsapi-shim';
+import { usePromiseFactory } from '@deephaven/react-hooks';
+
+/**
+ * Creates and subscribes to a `selectDistinct` derived table and unsubscribes
+ * on unmount.
+ * @param table The table to call `selectDistinct` on.
+ * @param columnNames The list of column names to pass to `selectDistinct`.
+ */
+export default function useSelectDistinctTable(
+  table: Table | TreeTable | null,
+  ...columnNames: string[]
+) {
+  const selectDistinct = useCallback(
+    async () => table?.selectDistinct(table.findColumns(columnNames)) ?? null,
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [table, ...columnNames]
+  );
```

Comment:
Guess you needed to disable the exhaustive deps because you're spreading the columnNames... any reason to not just pass in column names as an immutable array, and then use that?

Reply:
hmm, that hadn't occurred to me, but it seems that puts more burden on the caller vs just being able to

`useSelectDistinctTable(table, 'blah', 'bleh')`

I can add a comment to clarify unless you feel strongly about passing in an array

Reply:
Yea I don't feel too strongly about it.